internal/manifests/config/build.go

```diff
@@ -3,24 +3,91 @@ package config
 import (
 	"bytes"
 	"embed"
+	"fmt"
 	"html/template"
 	"io"
-)
-
-const (
+	"strings"
 
-	// TempoConfigFileName is the name of the config file in the configmap.
-	TempoConfigFileName = "config.yaml"
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+	"github.com/os-observability/tempo-operator/internal/manifests/manifestutils"
 )
 
 var (
 	//go:embed tempo-config.yaml
 	tempoConfigYAMLTmplFile embed.FS
 	tempoConfigYAMLTmpl     = template.Must(template.ParseFS(tempoConfigYAMLTmplFile, "tempo-config.yaml"))
+
+	//go:embed tempo-overrides.yaml
+	tempoTenantsOverridesYAMLTmplFile embed.FS
+	tempoTenantsOverridesYAMLTmpl     = template.Must(template.ParseFS(tempoTenantsOverridesYAMLTmplFile, "tempo-overrides.yaml"))
 )
 
-// Build builds a tempo configuration.
-func buildConfiguration(opts options) ([]byte, error) {
+func s3FromParams(params Params) s3 {
+	s3Insecure := false
+	s3Endpoint := params.S3.Endpoint
+	if strings.HasPrefix(s3Endpoint, "http://") {
+		s3Insecure = true
+		s3Endpoint = strings.TrimPrefix(s3Endpoint, "http://")
+	} else if !strings.HasPrefix(s3Endpoint, "https://") {
+		s3Insecure = true
+	} else {
+		s3Endpoint = strings.TrimPrefix(s3Endpoint, "https://")
+	}
```

Comment:
If the `s3Endpoint` comes with a `https://` prefix, we end up with `s3Insecure = true` but the prefix is not trimmed, right?

```suggestion
	s3Insecure := !strings.HasPrefix(s3Endpoint, "https://")
	s3Endpoint := params.S3.Endpoint
	if strings.HasPrefix(s3Endpoint, "http://") {
		s3Endpoint = strings.TrimPrefix(s3Endpoint, "http://")
	} else {
		s3Endpoint = strings.TrimPrefix(s3Endpoint, "https://")
	}
```

Would that solve the issue?

Reply:
I actually copied this code as it it. let me think about it a little bit :)

Reply:
right!, I've already commit your suggestion! nice catch, thanks!